src/connection/Connection.ts

```diff
@@ -106,7 +111,11 @@ export class Connection implements IConnection {
       else valueToSend = value;
     });
 
-    if (valueToSend !== {}) messages.push({ topic, propertyName: current, value: valueToSend });
+    // If the message is an object, and it's empty, it makes no sense to send it
+    // All other messages (e.g. non-empty objects or primitives) must be sent
+    if (!(typeof valueToSend === 'object' && Object.keys(valueToSend).length == 0)) {
```

Comment:
@fabioimpe do you think it's ok now, with an explicit check on the object and its length?
I think now is more explicit that:
- if it's an empty object, skip sending the message
- if it's a non-empty object, or any other value (primitive) then send it.

I thing it's rather an edge-case, which probably happens if the `properties` array is empty (so `forEach` statement is never executed) or `value` at line 111 is an empty object.

Reply:
Yes I think now is cleaner.
Maybe you can think about extracting the condition to be more readable, but this seems the right way 👍